plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1693,21 +1674,28 @@ public void testJsonb()
+                .addRoundTrip("hstore", "hstore(ARRAY['key1','value1','key2','value2','key3','value3'])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1','key2','key3'], ARRAY['value1','value2','value3']) AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "hstore(ARRAY['key1',' \" ','key2',' '' ','key3',' ]) '])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1','key2','key3'], ARRAY[' \" ',' '' ',' ]) ']) AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "hstore(ARRAY['key1',null])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1'], ARRAY[null]) AS MAP(VARCHAR, VARCHAR))")
                 .execute(getQueryRunner(), postgresCreateAndInsert("postgresql_test_hstore"));
-        hstoreTestCases(varcharMapDataType())
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("hstore", "NULL", createMapType(VARCHAR, VARCHAR), "CAST(NULL AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "MAP()", createMapType(VARCHAR, VARCHAR), "CAST(MAP() AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "MAP(ARRAY['key1'], ARRAY['value1'])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1'], ARRAY['value1']) AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "MAP(ARRAY['key1','key2','key3'], ARRAY['value1','value2','value3'])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1','key2','key3'], ARRAY['value1','value2','value3']) AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "MAP(ARRAY['key1','key2','key3'], ARRAY[' \" ',' '' ',' ]) '])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1','key2','key3'], ARRAY[' \" ',' '' ',' ]) ']) AS MAP(VARCHAR, VARCHAR))")
+                .addRoundTrip("hstore", "MAP(ARRAY['key1'], ARRAY[null])", createMapType(VARCHAR, VARCHAR), "CAST(MAP(ARRAY['key1'], ARRAY[null]) AS MAP(VARCHAR, VARCHAR))")
                 .execute(getQueryRunner(), postgresCreateTrinoInsert("postgresql_test_hstore"));
     }
 
-    private DataTypeTest hstoreTestCases(DataType<Map<String, String>> varcharMapDataType)
+    private Type createMapType(Type keyType, Type valueType)
     {
-        return DataTypeTest.create()
-                .addRoundTrip(varcharMapDataType, null)
-                .addRoundTrip(varcharMapDataType, ImmutableMap.of())
-                .addRoundTrip(varcharMapDataType, ImmutableMap.of("key1", "value1"))
-                .addRoundTrip(varcharMapDataType, ImmutableMap.of("key1", "value1", "key2", "value2", "key3", "value3"))
-                .addRoundTrip(varcharMapDataType, ImmutableMap.of("key1", " \" ", "key2", " ' ", "key3", " ]) "))
-                .addRoundTrip(varcharMapDataType, Collections.singletonMap("key1", null));
+        return getQueryRunner().getTypeManager().getType(mapType(keyType.getTypeSignature(), valueType.getTypeSignature()));
```

Comment:
I notice we only use `map(VARCHAR, VARCHAR)` so instead I'm extracting a variable and removing that helper function (since it's not used anywhere else).